### ISSUE TITLE
Question: How are we going to use statuses in `Dart`/`Flutter`?

### ISSUE THREAD

**Issue by nelsonic** [dwyl/statuses#10]
As noted in https://github.com/dwyl/statuses/issues/4#issuecomment-1179609113 I feel that **`JSON`** is a more versatile format for storing the `statuses` list than a `List` of `Elixir` `Map` or `Struct` ... 

As per https://github.com/dwyl/product-roadmap/issues/40 we are building a **`Flutter`** (`Native` Mobile App) version of the **`MVP`** that will be **_offline_ first**. We need `statuses` in `Dart`-land so that it can be included in the _compiled_ `Native` App. 

We could _either_ create a [`Struct`](https://api.flutter.dev/flutter/dart-ffi/Struct-class.html) in `Dart` and _duplicate_ the `statuses` 
and thus have to _maintain_ them in multiple places ... **`!DRY`**

***or***

use **`JSON`**.

**Comment by nelsonic** [comment]
@LuchoTurtle the reason I assigned this to you is that you can _help_ us understand the best way of using the `statuses` in `Flutter`-land. Ideally we would only manage a _single_ source of truth for this. i.e. we don't _want_ to create a separate `Dart` package ... 💭

**Comment by LuchoTurtle** [comment]
`Dart`, being quite close to `Typescript` when it comes to syntax, I wager having a `JSON` format is suitable to this use case, as you've stated. Parsing `json` files remain the best option for data exchange and parsing it comes out of the box with the language.

> No need to go for formats like [`json-dl`](https://json-ld.org/), there's no official support so let's keep it to the basics.

In Flutter, parsing local `json` files is as simple as 

```dart
final String response = await rootBundle.loadString('assets/sample.json');
final data = await json.decode(response);
```

and declaring its location in the `pubspec.yaml` file.

```yaml
flutter:
assets:
- assets/sample.json
```

However, I have a question regarding the `DRY` principle and having a `json` file detailing each status will affect this repo. 
It's clear you don't want to have yet another package for `Dart` to have these status. By having a single `json` file that is parsable, will it deprecate the Elixir library in this repo?

Does it make sense to have a `Dart` package in this repo that makes use of this `json` file, simply parsing it and then using it on whatever `Flutter` app we need it? Does it make sense for us to do the same for the `Elixir` library?

**Comment by nelsonic** [comment]
The `parse` a `JSON` file and use it is _exactly_ the approach taken in https://github.com/dwyl/quotes which is a _basic_ showcase of a module/package available in two languages/ecosystems `JavaScript` (`NPM`) and `Elixir` (`Hex.pm`).

My question stems from the fact that we did have `JSON` in _this_ repo `before` ...
and it was removed int: https://github.com/dwyl/statuses/pull/3/files 💭 
As much as I'd like to have a `Struct` that would allow type-checking 
But think that can be achieved in a statically-typed language even _with_ the `JSON` approach ...

In `Elixir` this appears to be the approach: https://elixirforum.com/t/how-to-decode-a-json-into-a-struct-safely/14331
And in `Dart` we can define a `class`: https://www.technicalfeeder.com/2022/02/parse-json-into-a-dart-class-object/?utm_content=cmp-true ⚠️ don't open that without an Ad Blocker! 

What I would _like_ to have is a compile-time check that a given status `int` is defined. 
i.e. we cannot accidentally have `status: 5000` if it's not in the `JSON` file. 🚫
